app/delivery/send_to_providers.py

```diff
@@ -123,23 +123,25 @@ def send_email_to_provider(notification):
                     # Throw error so celery will retry in sixty seconds
                     raise MalwarePendingException
 
-            try:
-                req = urllib.request.Request(personalisation_data[key]['document']['direct_file_url'])
-                with urllib.request.urlopen(req) as response:
-                    buffer = response.read()
-                    mime_type = magic.from_buffer(buffer, mime=True)
-                    if mime_type == 'application/pdf':
+            if sending_method == 'attach':
+                try:
+
+                    req = urllib.request.Request(personalisation_data[key]['document']['direct_file_url'])
+                    with urllib.request.urlopen(req) as response:
+                        buffer = response.read()
                         filename = personalisation_data[key]['document'].get('filename')
                         attachments.append({
-                            "name": filename or f'{key}.pdf',
+                            "name": filename,
                             "data": buffer
                         })
-            except Exception:
-                current_app.logger.error(
-                    "Could not download and attach {}".format(personalisation_data[key]['document']['direct_file_url'])
-                )
-
-            personalisation_data[key] = personalisation_data[key]['document']['url']
+                except Exception:
+                    current_app.logger.error(
+                        "Could not download and attach {}".format(
+                            personalisation_data[key]['document']['direct_file_url'])
+                    )
+                del personalisation_data[key]
```

Comment:
Do we have a test proving this line is here? I may have missed it, asking you for confirmation.

Reply:
I don't believe so. I put this line in as a cleanup knowing the purpose of this personalisation is done before template rendering. Extra personalisation keys when generating a template do not raise errors even prior to this PR, it's only when a template has variables specified and keys are missing is when errors are raised at render time.